mysql-test/suite/galera/t/MDEV-32738.test

```diff
@@ -0,0 +1,73 @@
+#
+# MDEV-32738: Don't BF-abort high-priority transaction waiting on a lock in InnoDB.
+#
+# DML transactions on FK-child tables also get table locks on FK-parent tables.
+# If there is a DML transaction holding such a lock, and a TOI transaction arrives,
+# the latter BF-aborts the former and puts itself into a waiting state.
+# If at this moment another DML transaction on FK-child table arrives, it doesn't
+# check that the transaction waiting on a parent table lock is TOI, and it
+# erroneously BF-aborts the waiting TOI transaction.
+#
+# The fix: check that the waiting transaction is TOI and in this case roll back
+# the incoming DML transaction in InnoDB.
+#
+# The test runs with autocommit statement retry disabled.
+#
+
+--source include/galera_cluster.inc
+--source include/have_debug_sync.inc
+--source include/have_debug.inc
+
+--connect con1,127.0.0.1,root,,test,$NODE_MYPORT_1
+--connect con2,127.0.0.1,root,,test,$NODE_MYPORT_1
```

Comment:
Maybe the connection `con2` could be named in some other way? I got confused because I thought it was going to `node_2`.